src/mail.sh

```diff
@@ -342,6 +342,10 @@ function find_commit_references()
     parsed=''
   done <<< "$(git rev-parse -- $args 2> /dev/null)"
 
+  if [[ -f "$i" || -d "$i" ]]; then
+    return 0
+  fi
```

Comment:
I am a little confused, where does `i` come from? I don't see it declared anywhere in this function neither see any relation with `args`, which is the variable that stores the files/directory arguments. It seems to work, as `kw mail --send` and `kw mail --send *.patch` work as intended (i.e., the first sends the last commit as a patch, and the second sends the patch files as patches **without** also including the last commit), but it isn't clear how. Please, comment your reasoning behind this, in case I am mistaken about it being wrong.

There are also two points:

1. As `kw mail` "enhances" `git send-email` we have an opportunity to check if the arguments are valid patch files. So I think we could do some kind of loop to check if each argument is a patch file (in case they are files) or a directory containing patch files (in case it is a directory).
2. I think the check to see if the arguments are patch files/directories should come after the below check (`if [[ -n "$commit_range" ]]; then`), as this last one is related with the while loop above (`while read -r arg; do`). I am saying this, because this function now finds commit references **or** patch files/directories, so it should check those separately.